src/aequitas/flow/methods/preprocessing/unawareness.py

```diff
@@ -121,19 +123,22 @@ def fit(self, X: pd.DataFrame, y: pd.Series, s: Optional[pd.Series]) -> None:
         s : pandas.Series
             Protected attribute vector.
         """
-        self.correlations = pd.Series(index=X.columns)
+        super().fit(X, y, s)
+
+        self.logger.info("Calculating feature correlation with sensitive attribute.")
 
-        for col in X.columns:
-            if X[col].dtype.name == "category":
-                self.correlations[col] = self._cramerv(
-                    s.astype("category").values, X[col].values
-                )
-            else:
-                self.correlations[col] = self._correlation_ratio(
-                    s.astype("category").values, X[col].values
-                )
+        if self.strategy == "correlation":
+            self.scores = pd.Series(index=X.columns)
+            for col in X.columns:
+                if X[col].dtype.name == "category":
+                    self.scores[col] = self._cramerv(s.values, X[col].values)
+                else:
+                    self.scores[col] = self._correlation_ratio(s.values, X[col].values)
 
-        self.correlations = self.correlations.sort_values(ascending=False)
+        elif self.strategy == "featureselection":
```

Comment:
I would prefer to see this in the `__init__` method, to raise it sooner rather than later!